tutorial/creating_packages/other_types_of_packages/header_only_packages.rst

```diff
@@ -1,2 +1,251 @@
+
+
+A header-only library is composed only of header files. That means a consumer doesn't link with any library but
+includes headers, so we need only one binary configuration for a header-only library.
+
+In the :ref:`Create your first Conan package
+<creating_packages_create_your_first_conan_package>` section, we learned about the settings, and how building the
+recipe applying different ``build_type`` (Release/Debug) generates a new binary package.
+
+As we only need one binary package, we don't need to declare the `settings` attribute.
+This is a basic recipe for a header-only recipe:
+
+.. code-block:: python
+   :caption: conanfile.py
+
+
+    from conan import ConanFile
+    from conan.tools.files import copy
+
+
+    class SumConan(ConanFile):
+        name = "sum"
+        version = "0.1"
+        # No settings/options are necessary, this is header only
+        exports_sources = "include/*"
+        # We can avoid copying the sources to the build folder in the cache
+        no_copy_source = True
+
+        def package(self):
+            # This will also copy the "include" folder
+            copy(self, "*.h", self.source_folder, self.package_folder)
+
+Our header-only library is this simple function that sums two numbers:
+
+
+.. code-block:: cpp
+   :caption: include/sum.h
+
+    inline int sum(int a, int b){
+        return a + b;
+    }
+
+
+The folder `examples2/tutorial/creating_packages/other_packages/header_only` in the cloned project contains a ``test_package``
+folder with an example of an application consuming the header-only library. So we can run a ``conan create .`` command
+to build the package and test the package:
+
+.. code-block:: bash
+
+    $ conan create .
+    ...
+    [ 50%] Building CXX object CMakeFiles/example.dir/src/example.cpp.o
+    [100%] Linking CXX executable example
+    [100%] Built target example
+
+    -------- Testing the package: Running test() ----------
+    sum/0.1 (test package): Running test()
+    sum/0.1 (test package): RUN: ./example
+    1 + 3 = 4
+
+After running the ``conan create`` a new binary package is created for the header-only library, and we can see how the
+``test_package`` project can use it correctly.
+
+We can list the binary packages created running this command:
+
+.. code-block:: bash
+
+    $ conan list packages sum/0.1#latest
+    Local Cache:
+        sum/0.1#154c10cda76635e3ca85bcd7e21c2de9:da39a3ee5e6b4b0d3255bfef95601890afd80709
+
+We get one package with the package ID ``da39a3ee5e6b4b0d3255bfef95601890afd80709``.
+Let's see what happen if we run the ``conan create`` but specifying ``-s build_type=Debug``:
+
+.. code-block:: bash
+
+    $ conan create . -s build_type=Debug
+    $ conan list packages sum/0.1#latest
+    Local Cache:
+        sum/0.1#154c10cda76635e3ca85bcd7e21c2de9:da39a3ee5e6b4b0d3255bfef95601890afd80709
+
+Even in the ``test_package`` executable is built for Debug, we get the same binary package for the header-only library.
+
```

Comment:
Maybe here we can comment again that this is because we did not declare the settings in the recipe, because people may not be aware that this is the result from that